dartagnan/src/main/java/com/dat3m/dartagnan/program/arch/aarch64/event/RMWStoreExclusiveStatus.java

```diff
@@ -48,13 +52,20 @@ public String toString() {
     }
 
     @Override
-    protected BoolExpr encodeExec(Context ctx){
+    protected BooleanFormula encodeExec(SolverContext ctx){
+        BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
+        IntegerFormulaManager imgr = ctx.getFormulaManager().getIntegerFormulaManager();
+
         int precision = register.getPrecision();
-		BoolExpr enc = ctx.mkAnd(
-                ctx.mkImplies(storeEvent.exec(), ctx.mkEq(regResultExpr, new IConst(BigInteger.ZERO, precision).toZ3Int(this, ctx))),
-                ctx.mkImplies(ctx.mkNot(storeEvent.exec()), ctx.mkEq(regResultExpr, new IConst(BigInteger.ONE, precision).toZ3Int(this, ctx)))
+		BooleanFormula enc = bmgr.and(
+				bmgr.implication(storeEvent.exec(), imgr.equal(
+						(IntegerFormula)regResultExpr, 
+						(IntegerFormula)new IConst(BigInteger.ZERO, precision).toIntFormula(this, ctx))),
+				bmgr.implication(bmgr.not(storeEvent.exec()), imgr.equal(
+						(IntegerFormula)regResultExpr, 
+						(IntegerFormula)new IConst(BigInteger.ONE, precision).toIntFormula(this, ctx)))
```

Comment:
This code will break if precision > 0 since toIntFormula will return a BitVectorFormula that can't be cast to IntegerFormula. Again, a Helper method that can construct equality formulas might be good.